src/algorithms/libs/beidou_b1c_signal_processing.cc

```diff
@@ -873,7 +873,7 @@ void beidou_b1c_code_gen_complex_sampled_boc(gsl::span<std::complex<float>> _des
                 }
             else
                 {
-                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 1.0 - 2.0 * imaginary_code_span_pilot[_codeLength - 1]);  //repeat the chip -> upsample
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 1.0 - 2.0 * imaginary_code_span_pilot[_codeValueIndex - 1]);  //repeat the chip -> upsample
```

Comment:
Still, inconsistencies shown here, why `[_codeValueIndex - 1]` in the pilot but not in the data component?